webapp/test/test_updates.py

```diff
@@ -169,3 +169,72 @@ def test_process_list_empty_json(self):
             # NOTE: use copy of dict with json input, because process_list changes this dict
             self.updates_api.process_list(1, UPDATES_JSON_EMPTY.copy())
         assert "'package_list' is a required property" in str(context)
+
+class TestSplayTree(TestBase):
+    """ Set of tests for Splay tree implementation. """
+    hot_cache = HotCache()
+    test_nevra = []
+
+    @pytest.fixture(autouse=True)
+    def setup_dummy_data(self):
+        """ Create list with dummy nevra data for the tree. """
+        self.test_nevra.append({"key": 1, "content": {"response": "nevra1"}})
+        self.test_nevra.append({"key": 2, "content": {"response": "nevra2"}})
+        self.test_nevra.append({"key": 3, "content": {"response": "nevra3"}})
+        self.test_nevra.append({"key": 4, "content": {"response": "nevra4"}})
+        self.test_nevra.append({"key": 5, "content": {"response": "nevra5"}})
+        self.test_nevra.append({"key": 6, "content": {"response": "nevra6"}})
+
+    @pytest.fixture
+    def setup_small_tree(self):
+        """ Fixture creates small tree for testing. """
+        self.hot_cache.insert(self.test_nevra[2]["key"], self.test_nevra[2]["content"])
+        self.hot_cache.insert(self.test_nevra[0]["key"], self.test_nevra[0]["content"])
+        self.hot_cache.insert(self.test_nevra[1]["key"], self.test_nevra[1]["content"])
+
+    @pytest.fixture
+    def setup_huge_tree(self):
```

Comment:
This design requires to insert method name as an argument. But it doesn't return any value. In this case it's better to create simple helper method `_setup_huge_tree` and call it at the start of touched tests.